src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String NOT_FOUND_INDEX_MESSAGE = "Not found index: ";
+    private static final String NOT_FOUND_VALUE_MESSAGE = "Not found element with value: ";
+    private Object[] storage = new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        increaseStorage();
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size++;
+        validateIndex(index);
```

Comment:
```suggestion
        validateIndex(index);
        size++;
```

Reply:
firstly you should validate index and then increment size :)

Reply:
Very good advices, thank you!